samples/dnn/text_detection.cpp

```diff
@@ -20,65 +20,104 @@
     CommandLineParser parser(argc, argv, keys);
-    parser.about("Use this script to run TensorFlow implementation (https://github.com/argman/EAST) of "
-                 "EAST: An Efficient and Accurate Scene Text Detector (https://arxiv.org/abs/1704.03155v2)");
-    if (argc == 1 || parser.has("help"))
-    {
+    parser.about("Text Detection and Recognition using OpenCV"
+                 "Example: ./example_dnn_text_detection -rmp=<path to ResNet_CTC.onnx> -dmp=<path to DB_IC15_resnet50.onnx");
+
+    // Display help message if no arguments are provided or 'help' is requested
+    if (argc == 1 || parser.has("help")) {
         parser.printMessage();
         return 0;
     }
 
-    float confThreshold = parser.get<float>("thr");
-    float nmsThreshold = parser.get<float>("nms");
-    int width = parser.get<int>("width");
-    int height = parser.get<int>("height");
+    // Parsing command-line arguments
+    String detModelPath = parser.get<String>("detModelPath");
+    String detModelConfig = parser.get<String>("detModelConfig");
+    String recModelPath = parser.get<String>("recModelPath");
+    int height = parser.get<int>("inputHeight");
+    int width = parser.get<int>("inputWidth");
     int imreadRGB = parser.get<int>("RGBInput");
-    String detModelPath = parser.get<String>("detModel");
-    String recModelPath = parser.get<String>("recModel");
     String vocPath = parser.get<String>("vocabularyPath");
+    float binThresh = parser.get<float>("binaryThreshold");
+    float polyThresh = parser.get<float>("polygonThreshold");
+    double unclipRatio = parser.get<double>("unclipRatio");
+    uint maxCandidates = parser.get<uint>("maxCandidate");
+    float confThreshold = parser.get<float>("thr");
+    float nmsThreshold = parser.get<float>("nms");
 
-    if (!parser.check())
-    {
+    // Ensuring the provided arguments are valid
+    if (!parser.check()) {
         parser.printErrors();
         return 1;
     }
 
-    // Load networks.
-    CV_Assert(!detModelPath.empty() && !recModelPath.empty());
-    TextDetectionModel_EAST detector(detModelPath);
-    detector.setConfidenceThreshold(confThreshold)
-            .setNMSThreshold(nmsThreshold);
-
-    TextRecognitionModel recognizer(recModelPath);
+    // Asserting detection model path is provided
+    CV_Assert(!detModelPath.empty());
+
+    std::vector<std::vector<Point>> detResults;
+    // Reading the input image
+    Mat frame = imread(samples::findFile(parser.get<String>("input")));
+
+    // Initializing and configuring the text detection model based on the provided config
+    if (detModelConfig == "east") {
+        // EAST Detector initialization
+        TextDetectionModel_EAST detector(detModelPath);
+        detector.setConfidenceThreshold(confThreshold)
+                .setNMSThreshold(nmsThreshold);
+        // Setting input parameters specific to EAST model
+        detector.setInputParams(1.0, Size(width, height), Scalar(123.68, 116.78, 103.94), true);
```

Comment:
The only generic parameter is `Size(width, height)`. Please update `TextDetectionModel_EAST_Impl` constructor with the rest of preprocessing options so in sample you just need to call `detector.setInputSize()`

 https://github.com/opencv/opencv/blob/f2c3d4dfe3843898d969a45e43d981e8af967bd5/modules/dnn/src/model.cpp#L1115

Reply:
@dkurt The other parameters can be generic if someone trains EAST model on custom dataset. We can add the default values in the constructor, but I propose that the call should remain setInputParams in the sample to show its usage